src/app/pages/signing/signing.component.html

```diff
@@ -169,3 +178,105 @@ <h5>Emojis</h5>
+          That means that in the future if the link stops working then so will the message you wrote that contains the
+          image.</strong> For example, Google Photos is inconsistent in whether its embed links work for others. Other services such
+          as Imgur or Pinterest can be used to store any images for sharing. Virtrolio is not endorsing any services
+          mentioned here, merely making you aware of options available to them.</p>
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<!-- Embed Photo Modal -->
+<div class="modal fade" id="embed-photo" tabindex="-1" role="dialog" aria-labelledby="embed-photo-label"
+     aria-hidden="true">
+  <div class="modal-dialog modal-dialog-centered modal-lg" id="embed-photo-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="embed-photo-title">Say cheese! Let's add an image!</h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body d-md-flex p-4" id="embed-photo-div">
+        <div class="col-md-6 my-3 px-lg-5 px-3 vertical-center">
+          <div class="mb-3">
+            <span class="mr-3"><strong>Add a link to a photo!</strong></span>
+            <button class="btn-sm btn-primary" data-toggle="modal" data-target="#embed-modal">
+              Need help?
+            </button>
+          </div>
+          <div class="input-group">
+            <label for="embed-link-field" style="display:none">Insert embed link</label>
+            <input class="form-control" type="text" placeholder="Put your link here!"
+                   [(ngModel)]='embedLink' id="embed-link-field">
+          </div>
+          <div class="mt-3">
+            <label for="width-slider" class="mx-2">Image Size (%)</label>
+            <input [(ngModel)]="imageWidth" class="mx-2" type="range" id="width-slider" min="0" max="100">
+          </div>
+          <div class="my-3">
+            <strong>Your image code</strong><br>
+            <small>Copy this into your message and check the 'Preview'!</small>
+            <div class="input-group my-3">
+              <input #EmbedCode type="text" class="form-control"
+                     value="<img src='{{embedLink}}' width={{imageWidth}}%>">
+              <div class="input-group-append">
+                <button class="btn btn-primary" type="button" (click)="copyLink(EmbedCode)">{{copyButtonText}}</button>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div class="col-md-6 mb-3 mb-md-0" id="embed-image-preview align-items-center">
+          <p><strong>Image Preview</strong></p>
```

Comment:
The image preview should be centred in the box, and it should fill the entire box. Ideally, the box would resize to the aspect ratio of the image with a limitation on the max dimensions.

Reply:
That's not what I was going for, and that wouldn't be as helpful. Left aligning with the box being a constant width shows the user what the image will actually look like within their message (I wouldn't want them to have it be centered and pretty looking in the preview only for them to paste it into their message and have it look different.